cdc/cdc.cc

```diff
@@ -218,4 +219,195 @@ db_context db_context::builder::build() {
     };
 }
 
+class transformer final {
+public:
+    struct map_cmp {
+        template <typename Func>
+        bool cmp(const std::pair<net::inet_address, unsigned int>& a,
+                 const std::pair<net::inet_address, unsigned int>& b,
+                 Func&& func) const {
+            auto&& a_ip = func(a.first).ip;
+            auto&& b_ip = func(b.first).ip;
+            return a_ip < b_ip || (a_ip == b_ip && a.second < b.second);
+        }
+        bool operator()(const std::pair<net::inet_address, unsigned int>& a,
+                        const std::pair<net::inet_address, unsigned int>& b) const {
+            if (a.first.is_ipv4()) {
```

Comment:
This is too clever for me :-) what is this operator() supposed to return - equality? But then, if a is ipv4, and b is NOT ipv4, why do we return true?

Reply:
This is not equality. This is used in std::map so it has to return true if a < b.
If a is ipv4 and b is ipv6, the comparator returns true because I ordered all ipv4 addresses before all ipv6 addresses. Does it make sense?

Reply:
Yes, it makes sense, but please add a comment saying this - that this returns true if a < b, as std::map requires, and that you order ipv4 addresses before ipv6 addresses. It's hard to guess and there are many other possibilities on what this "cmp" function does.

Reply:
The traditional name for such a comparator is less(). But why can't you use inet_type's comparator?

Also (this is a question for where this is used) it's better to use an unordered_map when possible.

Reply:
I changed the map type to unordered_map so this comparator goes away.